src/execution/window_executor.cpp

```diff
@@ -812,7 +817,7 @@ WindowExecutorGlobalState::WindowExecutorGlobalState(const WindowExecutor &execu
 
 	auto types = payload_chunk.GetTypes();
 	if (!types.empty()) {
-		payload_collection.Initialize(Allocator::Get(executor.context), types);
+		payload_collection.Initialize(Allocator::Get(executor.context), types, payload_count);
```

Comment:
Hmm this is only used by `WindowValueExecutor` and friends, so maybe it needs to move there? Otherwise we will be accumulating data that we don't need. Since the aggregate accelerators do this independently, it seems like it would reduce memory consumption a fair bit. Especially since `WindowConstantAggregator` doesn't need to accumulate at all.